client.go

```diff
@@ -603,12 +603,10 @@ func (c *Client) runConfirmsConsumer(confirms chan amqp.Confirmation, returns ch
 
 			request, ok := c.requestsMap.GetByDeliveryTag(confirm.DeliveryTag)
 			if !ok {
-				// This could happen if we stop waiting for requests to return due
-				// to a timeout. But since confirmations are normally very fast that
-				// would mean that something isn't quite right on the amqp server.
-				// Unfortunately there isn't any way of getting more information
-				// than the delivery tag from a confirmation.
-				c.logger.Error(
+				// Responses can arrive before the confirmation so it's not an
+				// error if the request is removed from the requestsMap.
+				// See: https://www.rabbitmq.com/docs/confirms#publisher-confirms-latency
```

Comment:
Would this be an accurate description of what's happening?

```suggestion
				// Our AMQP consumer may receive RPC response deliveries before our
				// AMQP publisher receives a confirmation for the corresponding RPC
				// request delivery. In such cases, the request may have already
				// been removed from the requestsMap. This is expected behavior, not
				// an error.
```

It might be obvious for someone with experience working in the library that "responses" and "confirmations" refer to concepts at different layers (IIUIC), i.e. RPC and AMQP, respectively. For someone like me, however, it helps having that distinction spelled out explicitly to understand the described scenario. What do you think?